File: Phase08-EFCore/SearchLib/SearchLib.Test/Utils/InvertedIndexTest.cs
using FluentAssertions;
using SearchLib.Model;
using SearchLib.Utils;
using System.Collections.Generic;
using System.Linq;
using Xunit;

namespace SearchLib.Test.Utils
{
    public class InvertedIndexTest
    {

        private InvertedIndex Index;

        public InvertedIndexTest()
        {
            SearchContext context = new SearchContext("localhost", "root", "", "test_searchlib_inverted_index");
            context.Database.EnsureCreated();
            Index = new InvertedIndex(context);
        }

        [Theory]
        [InlineData(new string[] { "..\\..\\..\\..\\SampleEnglishData\\57110", "..\\..\\..\\..\\SampleEnglishData\\58043" }, "have")]
        [InlineData(new string[] { "..\\..\\..\\..\\SampleEnglishData\\58043" }, "same")]
        public void AddWordTest(string[] documents, string word)
        {
            foreach (string document in documents)
            {
                Index.AddWord(new Document(document), word);
            }
            ISet<string> actualDocuments = Index.GetWordIndexes(word).Select(e => e.Path).ToHashSet();
            actualDocuments.Should().BeEquivalentTo(documents.Select(e => new Document(e).Path));
        }
    }
}

File: Phase01-Search/src/test/java/utils/InvertedIndexTest.java
package utils;

import model.Document;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.nio.file.Path;
import java.util.HashSet;
import java.util.Set;

class InvertedIndexTest {
    InvertedIndex invertedIndex;
    
    @BeforeEach
    void init() {
        invertedIndex = new InvertedIndex();
    }

    @Test
    void addWordTest() {
        invertedIndex.addWord(new Document(Path.of("SampleEnglishData/57110")), "have");
        invertedIndex.addWord(new Document(Path.of("SampleEnglishData/58043")), "have");
        invertedIndex.addWord(new Document(Path.of("SampleEnglishData/58043")), "same");
        Set<Document> firstExpectedDocuments = new HashSet<>();
        Set<Document> secondExpectedDocuments = new HashSet<>();
        firstExpectedDocuments.add(new Document(Path.of("SampleEnglishData/57110")));
        firstExpectedDocuments.add(new Document(Path.of("SampleEnglishData/58043")));
        secondExpectedDocuments.add(new Document(Path.of("SampleEnglishData/58043")));
        Assertions.assertAll(() -> {
            Assertions.assertArrayEquals(firstExpectedDocuments.toArray(), invertedIndex.getWordIndexes("have").toArray());
            Assertions.assertArrayEquals(secondExpectedDocuments.toArray(), invertedIndex.getWordIndexes("same").toArray());
        });
    }
}
